source/common/http/http1/codec_impl.cc

```diff
@@ -259,7 +261,13 @@ const ToLowerTable& ConnectionImpl::toLowerTable() {
 }
 
 ConnectionImpl::ConnectionImpl(Network::Connection& connection, http_parser_type type)
-    : connection_(connection) {
+    : connection_(connection), output_buffer_(Buffer::InstancePtr{new Buffer::OwnedImpl()},
+                                              [&]() -> void { this->onBelowLowWatermark(); },
+                                              [&]() -> void { this->onAboveHighWatermark(); }) {
+  uint32_t buffer_limit = connection.bufferLimit();
+  if (buffer_limit > 0) {
+    output_buffer_.setWatermarks(buffer_limit / 2, buffer_limit);
```

Comment:
The connection does so for TCP proxying to avoid watermark thrashing, where bytes in generally equals bytes out.

We could assert the same for HTTP as for body transfers with no transformations that may be the case, but for the bulk of read/write scenarios it's less likely.  Totally your call.